os/fs/driver/mtd/smart.c

```diff
@@ -4943,22 +4973,29 @@ static int smart_journal_checkin(FAR struct smart_struct_s *dev, journal_log_t *
 	ret = MTD_WRITE(dev->mtd, address, log_size, (FAR uint8_t *)log);
 	if (ret != log_size) {
 		fdbg("Checkin Error, write log data... physical address %u ret : %d\n", address, ret);
-		return -EIO;
+		ret = -EIO;
+		goto errout;
 	}
 
 	/* Verify written journal data */
 	ret = MTD_READ(dev->mtd, address, log_size, (FAR uint8_t *)&buf);
 	if (ret != log_size) {
 		fdbg("Checkin Error, read log data... physical address %u ret : %d\n", address, ret);
-		return -EIO;
+		ret = -EIO;
+		goto errout;
 	}
 
 	if (smart_validate_journal_crc(&buf) != OK) {
 		fdbg("Checkin Error, verify crc failed buf crc16 : %d calc crc : %d\n", UINT8TOUINT16(buf.crc16), smart_calc_journal_crc(&buf));
-		return -EIO;
+		ret = -EIO;
+		goto errout;
```

Comment:
When going to errout, `ret` will be overwritten by `smart_journal_checkout()`. So I think we don't need to assing `ret` as `-EIO`. How about setting that to errno?

Reply:
you're right, it will overwrite, I will fix it.
Anyway set_errno is not good choice, file system always set_errno in vfs level, not sub file system. Maybe kernel also same.